ansible/roles/yarn/templates/yarn-site.xml

```diff
@@ -37,6 +37,14 @@
   <property>
     <name>yarn.nodemanager.log.retain-seconds</name>
     <value>3600</value>
+  </property>
```

Comment:
@Kaali09 The yarn documentation says the following:
`This document gives an overview of NodeManager (NM) restart, a feature that enables the NodeManager to be restarted without losing the active containers running on the node. At a high level, the NM stores any necessary state to a local state-store as it processes container-management requests. When the NM restarts, it recovers by first loading state for various subsystems and then letting those subsystems perform recovery using the loaded state.`
We see that killing a particular job does not kill all the existing containers and hence there are issues related to data duplication. I am not sure if this property will solve the issue. Can you please elaborate why we need to make this configuration change?

Reply:
So far, identified 2 scenarios where the container processes are becoming orphan process. By Enabling this yarn.nodemanager.recovery.enabled flag, we can avoid the orphan process which occurring based on the below 1st scenario and same has been tested in demo environment which we created for sunbird workshop.

and for the 2nd scenario, as of now we dont have solution. 

Scenarios: 
1. If the node manager goes down, the process which are all running on the particular node becoming orphan process.
2. When deploying the new version samza app, the deploy job kill the previous version of samza app(currently running app), so related process which are running in yarn slaves should gets killed. But some times its not happening and those processes are becoming orphan process.